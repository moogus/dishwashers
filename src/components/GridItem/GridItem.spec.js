import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { useNavigate } from '@reach/router';

import { GridItem } from './index';

jest.mock('@reach/router');

describe('<GridItem/>', () => {
  const renderContainer = ({
    productId = '1234',
    price = 'Â£123',
    title = 'Hiya',
    image = 'a picture',
  } = {}) => {
    const props = {
      gridItem: {
        productId,
        price,
        title,
        image,
      },
    };
    return render(<GridItem {...props} />);
  };

  describe('when <GridItem/> is rendered', () => {
    const navigate = jest.fn();
    beforeEach(() => {
      jest.clearAllMocks();
      useNavigate.mockReturnValue(navigate);
    });

    test('component renders as expected', () => {
      const { container } = renderContainer();
      expect(container).toMatchSnapshot();
    });

    test('clicking the grid item navigates to the product page', () => {
      const { getByTestId } = renderContainer();
      const clickableGridItem = getByTestId('clickable-grid-item');
      userEvent.click(clickableGridItem);
      expect(navigate).toHaveBeenCalledWith('product/1234');
    });
  });
});
