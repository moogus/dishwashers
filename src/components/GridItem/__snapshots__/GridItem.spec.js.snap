// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridItemComponent/> when <GridItem/> is rendered component renders as expected 1`] = `
<div>
  <div
    class="sc-bdfBwQ gQwMDF col-md-3"
    data-testid="clickable-grid-item"
  >
    <div
      class="sc-gsTCUz wFoFg"
    >
      <div
        class="sc-dlfnbm gYlIvC row"
      >
        <div
          class="sc-hKgILt bEKIEI col"
        >
          <img
            alt="Hiya"
            class="sc-eCssSg knjoeH"
            src="a picture"
          />
        </div>
      </div>
      <div
        class="sc-jSgupP gylmqB"
      >
        <p
          class="sc-gKsewC ebGEqF"
        >
          Hiya
        </p>
      </div>
      <div
        class="sc-iBPRYJ kspsru"
      >
        <p
          class="sc-fubCfw gkHLwA"
        >
          Â£123
        </p>
      </div>
    </div>
  </div>
</div>
`;
